SparkleSpin/Controllers/HomeViewController.swift

```diff
@@ -10,8 +10,15 @@ import UIKit
 
 class HomeViewController: UIViewController {
 
-    override func viewDidLoad() {
-        super.viewDidLoad()
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        navigationController?.setNavigationBarHidden(true, animated: animated)
+        navigationItem.backBarButtonItem = UIBarButtonItem(title: "", style: .plain, target: nil, action: nil)
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        super.viewWillDisappear(animated)
+        navigationController?.setNavigationBarHidden(false, animated: animated)
```

Comment:
What's the reasoning for hiding the navigation bar in `viewWillAppear ` and then showing it again here?

Reply:
I wanted the navigation bar to become un-hidden when the Home screen disappeared so I wouldn't have to set `setNavigationBarHidden` to `false` in the `viewWillAppear` method of any other view controller in the app.

It's easier for me to remember to set `setNavigationBarHidden` to `true` in every view controller in which I don't want the navigation bar to appear than to remember to set it to `false` when I want to un-hide the navigation bar in virtually every other view controller.

Reply:
Maybe I'm missing something but I don't see any difference in the UI with or without this code, I commented out this two lines like this: 

```
override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        //navigationController?.setNavigationBarHidden(true, animated: animated)
        navigationItem.backBarButtonItem = UIBarButtonItem(title: "", style: .plain, target: nil, action: nil)
    }
    
    override func viewWillDisappear(_ animated: Bool) {
        super.viewWillDisappear(animated)
        //navigationController?.setNavigationBarHidden(false, animated: animated)
    }
```

And UI wise looks same is if those lines where not commented out.

Reply:
Maybe we can talk about this in our next session. I'm going to approve in the meantime, I'm just genuinely  curious 😃

Reply:
Ah, that's because the navigation bar is the same color as the main view in the Home view controller. I think it might be better to hide it still, just so there are no surprises if I change colors.

Reply:
ahh, that makes sense!